### ISSUE TITLE
OCI runtime exec failed: exec failed: unable to start container process: exec: "bash": executable file not found in $PATH: unknown

### ISSUE THREAD

**Issue by mvpwar** [argoproj/argo-cd#9605]
Terminal exec(v2.4.0-rc5) only support the docker image which entrypoint is "bash", like nginx. Could "sh" entrypoint (is widely used) be supported by stable version in the future?  THX!

**Terminal error messages:** 
OCI runtime exec failed: exec failed: unable to start container process: exec: "bash": executable file not found in $PATH: unknown

**F12:**
WebSocket is already in CLOSING or CLOSED state

**Comment by ohauer** [comment]
definitive vote for this!

**Comment by ohauer** [comment]
Not sure if this is the right code place where I found the reference, but it seems "sh" is already defined as validShell.
https://github.com/argoproj/argo-cd/blob/master/server/application/terminal.go#L219
Perhaps it helps most containers to change the order.

**Comment by crenshaw-dev** [comment]
Yeah, `sh` should already be supported. Maybe there's an issue in the logic that tries multiple shells.

**Comment by m-yosefpor** [comment]
I think changing the order is not a good solution, as many images provide both `sh` and `bash`, but colors, auto-completion, etc is working only with `bash`, and not `sh`.  I think the logic of trying other terminals should be fixed.

**Comment by m-yosefpor** [comment]
This is the argocd error message:

```
time="2022-06-20T11:28:12Z" level=info msg="terminal session starting" application=sth cluster=sth container=sth namespace=sth podName=sth-6db685f798-gr7ll userName=mohammad.yosefpor
E0620 11:28:12.808320       1 v2.go:105] EOF
time="2022-06-20T11:28:25Z" level=error msg="read message err: websocket: unexpected reserved bits 0x30"
E0620 11:28:25.602831       1 v2.go:105] websocket: unexpected reserved bits 0x30
time="2022-06-20T11:28:25Z" level=error msg="write message err: websocket: close sent"
E0620 11:28:25.625220       1 v2.go:151] websocket: close sent

```

Also it seems the session starts with sh, as I can see login, but the websocket is closed, so I can't type anything:

<img width="858" alt="image" src="https://user-images.githubusercontent.com/47300215/174593021-068da178-3f23-48e4-8230-1d78c5374f14.png">

**Comment by pf2rich** [comment]
This problem can be reproduced by starting the pod with an alpine or busybox image.

**Comment by pf2rich** [comment]
![image](https://user-images.githubusercontent.com/22682155/175508887-20355269-0fff-4c5d-87f8-ddbff9476a0f.png)
websocks auto input some string for alpine image

**Comment by m-yosefpor** [comment]
all the following 3 issues are the same:
- https://github.com/argoproj/argo-cd/issues/9643
- https://github.com/argoproj/argo-cd/issues/9641
- https://github.com/argoproj/argo-cd/issues/9605


I think we can follow in one thread.
